### ISSUE TITLE
Example idea: Monte Carlo

### ISSUE THREAD

**Issue by sbastrakov** [ComputationalRadiationPhysics/alpaka#610]
Since alpaka has support for both random number generators and atomic operations, it should be easy to make an example implementing Monte Carlo. To me two applications seem most fitting:
* [numerical integration](https://en.wikipedia.org/wiki/Monte_Carlo_integration) - this should be familiar to vast majority of potential users;
* [Black-Scholes](https://en.wikipedia.org/wiki/Black%E2%80%93Scholes_model) option pricing - this is probably a little bit more tricky, however it is also done rather extensively as a parallel programming example.

**Comment by tdd11235813** [comment]
could be helpful to have such a common example like MonteCarlo (Pi calculation ?), how to use different back-ends, random numbers and atomics (for reduction). At the moment examples does not have high priority, but maybe someone can write this example for learning alpaka.

**Comment by j-stephan** [comment]
We have created a Monte Carlo Compute Pi example for last year's CERN workshop. We should aim to pull this into the mainline alpaka code base.

**Comment by sbastrakov** [comment]
Yes, or something similar to it. Iirc when doing examples for that workshop we were also thinking to simplify the coding of existing alpaka examples to separate them from CI test logic and so be easier readable. So more in spirit of the workshop examples
